api/filter/http/rate_limit.proto

```diff
@@ -4,27 +4,28 @@ package envoy.api.v2.filter.http;
 
 import "google/protobuf/duration.proto";
 
-// HTTP rate limit filter configuration. The HTTP rate limit filter will
-// call the rate limit service when the request’s route or virtual host has
-// one or more rate limit configurations that match the filter stage
-// setting. The route can optionally include the virtual host rate limit
-// configurations. More than one configuration can apply to a request. Each
-// configuration results in a descriptor being sent to the rate limit
-// service. If the rate limit service is called, and the response for any
-// of the descriptors is over limit, a 429 response is returned.
+import "validate/validate.proto";
+
+// [#protodoc-title: Rate limit]
+// Rate limit :ref:`configuration overview <config_http_filters_rate_limit>`.
+
 message RateLimit {
   // The rate limit domain to use when calling the rate limit service.
-  string domain = 1;
+  string domain = 1 [(validate.rules).string.min_bytes = 1];
```

Comment:
out of curiosity, why are you using bytes instead of length here (and elsewhere in this PR for strings)?

Reply:
This has to do with the fact that doing min_len is tricky in C++, as it doesn't have native UTF-8 support. We just need non-empty for now, so this works.

Reply:
👌 got it, thanks